src/tracer.c

```diff
@@ -75,210 +75,189 @@ next_finfo(PROCESS_INFO *pi)
     return &(pi->finfo[++(pi->numfinfo)]);
 }
 
-char *
+char           *
 get_str_from_process(pid_t pid, void *addr)
 {
-    static char buf[PATH_MAX];
-    char *dest = buf;
-    union
-    {
-	long lval;
-	char cval[sizeof (long)];
+    static char     buf[PATH_MAX];
+    char           *dest = buf;
+    union {
+	long            lval;
+	char            cval[sizeof (long)];
     } data;
 
-    for (int i = 0;; i++)
-    {
+    size_t          i = 0;
+
+    do {
 	data.lval =
 		ptrace(PTRACE_PEEKDATA, pid, addr + i * sizeof (long), NULL);
-	for (int j = 0; j < sizeof (long); j++)
-	{
+	for (int j = 0; j < sizeof (long); j++) {
 	    *dest++ = data.cval[j];
 	    if (data.cval[j] == 0)
 		break;
 	}
-    }
+	++i;
+    } while (*dest);
     return strdup(buf);
 }
 
-/*====================================================================================*/
-
-#if 0
-// still TODO
+#define find(arr, size, pid)						\
```

Comment:
not a macro; a function getting a `pid` and returning a `PROCESS_INFO *` (the appropriate member of `pinfo`)